libdnf/advisory/advisory.cpp

```diff
@@ -0,0 +1,165 @@
+
+Advisory::Advisory(libdnf::rpm::SolvSack & sack, AdvisoryId id) : id(id), sack(sack.get_weak_ptr()) {}
+
+std::string Advisory::get_name() const {
+    const char * name;
+    Pool * pool = sack->p_impl->get_pool();
+
+    name = pool_lookup_str(pool, id.id, SOLVABLE_NAME);
+
+    if (strncmp(
+            libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX, name, libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX_LENGTH) !=
+        0) {
+        auto msg = fmt::format(
+            R"**(Bad libsolv id for advisory "{}", solvable name "{}" doesn't have advisory prefix "{}")**",
+            id.id,
+            name,
+            libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX);
+        throw RuntimeError(msg);
+    }
+
+    return std::string(name + libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX_LENGTH);
+}
+
+Advisory::Type Advisory::get_type() const {
+    const char * type;
+    Pool * pool = sack->p_impl->get_pool();
+    type = pool_lookup_str(pool, id.id, SOLVABLE_PATCHCATEGORY);
+
+    if (type == NULL) {
+        return Advisory::Type::UNKNOWN;
+    }
+    if (!strcmp(type, "bugfix")) {
+        return Advisory::Type::BUGFIX;
+    }
+    if (!strcmp(type, "enhancement")) {
+        return Advisory::Type::ENHANCEMENT;
+    }
+    if (!strcmp(type, "security")) {
+        return Advisory::Type::SECURITY;
+    }
+    if (!strcmp(type, "newpackage")) {
+        return Advisory::Type::NEWPACKAGE;
+    }
+
+    return Advisory::Type::UNKNOWN;
+}
+
+const char * Advisory::advisory_type_to_cstring(Type type) {
+    switch (type) {
+        case Type::ENHANCEMENT:
+            return "enhancement";
+        case Type::SECURITY:
+            return "security";
+        case Type::NEWPACKAGE:
+            return "newpackage";
+        case Type::BUGFIX:
+            return "bugfix";
+        default:
+            return NULL;
+    }
+}
+
+std::string Advisory::get_severity() const {
+    Pool * pool = sack->p_impl->get_pool();
+
+    //TODO(amatej): should we call SolvPrivate::internalize_libsolv_repo(solvable->repo);
+    //              before pool_lookup_str?
+    //              pool_lookup_str can return nullptr but std::string constructor cannot
+    //              accept nullptr (e.g. id = 0).
+    //              If so do this just once in solv::advisroy_private
+    return std::string(pool_lookup_str(pool, id.id, UPDATE_SEVERITY));
+}
+
+AdvisoryId Advisory::get_id() const {
+    return id;
+}
+
+std::vector<AdvisoryReference> Advisory::get_references(AdvisoryReferenceType ref_type) const {
+    Pool * pool = sack->p_impl->get_pool();
+
+    std::vector<AdvisoryReference> output;
+
+    Dataiterator di;
+    dataiterator_init(&di, pool, 0, id.id, UPDATE_REFERENCE, 0, 0);
+
+    for (int index = 0; dataiterator_step(&di); index++) {
+        dataiterator_setpos(&di);
+        const char * current_type = pool_lookup_str(pool, SOLVID_POS, UPDATE_REFERENCE_TYPE);
+
+        if (((ref_type & AdvisoryReferenceType::CVE) == AdvisoryReferenceType::CVE &&
```

Comment:
Note: In case that I will add new value into enum or I will pass incorrect value as ref_type the method will not recognize it.

Reply:
This is tied with: https://github.com/rpm-software-management/libdnf/pull/1043#discussion_r625916499